cmd/kubeadm/app/componentconfigs/config.go

```diff
@@ -63,6 +64,12 @@ func GetFromKubeProxyConfigMap(client clientset.Interface, version *version.Vers
 	// Read the ConfigMap from the cluster
 	kubeproxyCfg, err := apiclient.GetConfigMapWithRetry(client, metav1.NamespaceSystem, kubeadmconstants.KubeProxyConfigMap)
 	if err != nil {
+		// The Kube-Proxy config map may be non-existent, because the user has decided to manage it by themselves
+		// or to use other proxy solution. It may also be forbidden - if the kube-proxy phase was skipped we have neither
+		// the config map, nor the RBAC rules allowing join access to it.
+		if apierrors.IsNotFound(err) || apierrors.IsForbidden(err) {
+			return nil, nil
```

Comment:
my previous attempt had a warning here. should we have one?

Reply:
Depends on what we are trying to warn about. Not having a kube-proxy config map should be perfectly acceptable in my opinion, but we may warn the users, that they are on their own if they choose to manage the proxy addon by themselves.

Reply:
yeah i think we should warn something in the lines of that if the CM is missing or forbidden the users are on their own managing the addon.